client/payments/data/selectors.js

```diff
@@ -0,0 +1,10 @@
+export function retrievePaymentGateway( state, id ) {
+	const paymentGateway = state.paymentGateways.filter( ( gateway ) => {
```

Comment:
This is a lot of code thats not really needed, do you really need to coerce this to null? seems like undefined would be fine and you should use find not filter.

e.g.

```
return state.paymentGateways.find((gateway) => gateway.id === id);
```

if you really need to coerce to null then:

```
return state.paymentGateways.find((gateway) => gateway.id === id) || null;
```

Reply:
It actually returns `undefined` even if I set it to null. I will use undefined instead.